src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using AWS.Deploy.Common;
+using AWS.Deploy.Orchestration;
+using AWS.Deploy.Orchestration.CDK;
+using AWS.Deploy.Orchestration.Data;
+using AWS.Deploy.Orchestration.Utilities;
+using AWS.Deploy.Recipes;
+
+namespace AWS.Deploy.CLI.Commands
+{
+    public class GenerateDeploymentProjectCommand
+    {
+        private readonly IToolInteractiveService _toolInteractiveService;
+        private readonly IConsoleUtilities _consoleUtilities;
+        private readonly IOrchestratorInteractiveService _orchestratorInteractiveService;
+        private readonly ICdkProjectHandler _cdkProjectHandler;
+        private readonly ICDKManager _cdkManager;
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+        private readonly IDeploymentBundleHandler _deploymentBundleHandler;
+        private readonly ICommandLineWrapper _commandLineWrapper;
+        private readonly OrchestratorSession _session;
+
+        public GenerateDeploymentProjectCommand(
+            IToolInteractiveService toolInteractiveService,
+            IConsoleUtilities consoleUtilities,
+            IOrchestratorInteractiveService orchestratorInteractiveService,
+            ICdkProjectHandler cdkProjectHandler,
+            ICDKManager cdkManager,
+            IAWSResourceQueryer awsResourceQueryer,
+            IDeploymentBundleHandler deploymentBundleHandler,
+            ICommandLineWrapper commandLineWrapper,
+            OrchestratorSession session)
+        {
+            _toolInteractiveService = toolInteractiveService;
+            _consoleUtilities = consoleUtilities;
+            _orchestratorInteractiveService = orchestratorInteractiveService;
+            _cdkProjectHandler = cdkProjectHandler;
+            _cdkManager = cdkManager;
+            _awsResourceQueryer = awsResourceQueryer;
+            _deploymentBundleHandler = deploymentBundleHandler;
+            _commandLineWrapper = commandLineWrapper;
+            _session = session;
+        }
+
+        public async Task ExecuteAsync(string saveDirectory)
+        {
+            var orchestrator = GetOrchestrator();
+            var recommendations = await GenerateDeploymentRecommendations(orchestrator);
+            var selectedRecommendation = _consoleUtilities.AskToChooseRecommendation(recommendations);
+
+            if (string.IsNullOrEmpty(saveDirectory))
+                saveDirectory = GenerateDefaultSaveDirectory(_session.ProjectDefinition);
+
+            var directoryCreatedByDeployTool = ValidateSaveDirectory(saveDirectory);
+
+            var result = await CheckDirectoryUnderSourceControl(saveDirectory);
+            if (!result.Success)
+            {
+                _toolInteractiveService.WriteLine();
+                var YesNoResult = _consoleUtilities.AskYesNoQuestion("The target directory for saving the CDK " +
```

Comment:
nit: local variable should start lowercase.

Reply:
Fixed